File: leet_code/arrays/plus_one.js
const plusOne = arr => {
    if (arr[arr.length - 1] < 9) {
        arr[arr.length - 1] += 1;
        return arr;
    }

    for (let i = arr.length - 1; i >= 0; i--) {
        if (i === 0 && arr[i] === 9) {
            arr[i] = 0;
            arr.unshift(1);
        } else if ( arr[i] === 9) {
            arr[i] = 0;
        } else {
            arr[i] += 1;
            return arr;
        }
    }
    return arr;
};

File: leet_code/arrays/plus_one.rb
def plus_one(arr)
  if arr[-1] < 9
    arr[-1] += 1
    return arr
  end

  idx = arr.length - 1
  while idx >= 0
    if arr[idx] == 9 && idx == 0
      arr[idx] = 0
      arr.unshift(1)
      break
    elsif arr[idx] == 9
      arr[idx] = 0
    else
      arr[idx] += 1
      return arr
    end
    idx -= 1
  end
  arr
end
